lib/helpers/hearing_renderer.rb

```diff
@@ -0,0 +1,358 @@
+    end
+  end
+
+  def format_conversions(type_conversions)
+    type_conversions.map do |tc|
+      "Converted to #{tc['to_type']} from #{tc['from_type']} by #{tc['converted_by']} at #{tc['converted_at']}"
+    end
+  end
+
+  def add_original_type(original_type, type_conversions)
+    first_conversion = type_conversions[0]
+    first_conversion["from_type"] = original_type
+    type_conversions[0] = first_conversion
+
+    type_conversions
+  end
+
+  def get_appeal_history(appeal)
+    if appeal.versions.empty?
+      format_original_and_current_type(appeal)
+    else
+      type_conversions = get_appeal_type_conversions(appeal)
+      type_conversions_with_original_type = add_original_type(
+        appeal.readable_original_hearing_request_type,
+        type_conversions
+      )
+      text = format_original_and_current_type(appeal)
+      text += format_conversions(type_conversions_with_original_type)
+      text
+    end
+  end
+
+  def get_unscheduled_hearings(appeal)
+    regional_office = appeal.closest_regional_office
+    ro_label = regional_office.nil? ? COPY::UNKNOWN_REGIONAL_OFFICE : get_ro_label(regional_office)
+
+    open_schedule_hearing_tasks = appeal.tasks.open.of_type(:ScheduleHearingTask)
+    return "" if open_schedule_hearing_tasks.nil?
+
+    unscheduled_hearings = open_schedule_hearing_tasks.map do |sh_task|
+      unscheduled_hearing_label = "Unscheduled Hearing (SCH Task ID: #{sh_task.id}, RO queue: #{ro_label})"
+      unscheduled_note = show_pii ? sh_task.parent&.instructions : 'Show PII to see notes'
+
+      { unscheduled_hearing_label => ["Notes: #{unscheduled_note}"] }
+    end
+
+    unscheduled_hearings
+  end
+
+  def shared_appeal_children(appeal)
+    children = []
+    if appeal.appellant_is_not_veteran
+      children << "Appellant is not veteran - #{appeal&.appellant_relationship}"
+    end
+
+    children += appeal.hearings.map { |hearing| structure(hearing) }
+    children += get_unscheduled_hearings(appeal)
+
+    children << {
+      "Appeal History": get_appeal_history(appeal)
+    }
+    children
+  end
+
+  def legacy_appeal_details(legacy_appeal)
+    ["VACOLS ID: #{legacy_appeal.vacols_id}"]
+  end
+
+  def legacy_appeal_children(legacy_appeal)
+    shared_appeal_children(legacy_appeal)
+  end
+
+  def legacy_appeal_context(legacy_appeal)
+    legacy_appeal.veteran
+  end
+
+  def appeal_details(appeal)
+    ["UUID: #{appeal.uuid}"]
+  end
+
+  def appeal_children(appeal)
+    shared_appeal_children(appeal)
+  end
+
+  def appeal_context(appeal)
+    appeal.veteran
+  end
+
+  def scheduled_for(hearing)
+    ro_time = hearing.scheduled_for.in_time_zone(hearing.regional_office_timezone)
+    if ro_time != hearing.scheduled_for
+      "Scheduled for: #{readable_date(hearing.scheduled_for)} / #{ro_time.strftime('%I:%M%p %Z')}(RO time)"
+    end
+
+    "Scheduled for: #{readable_date(hearing.scheduled_for)}"
+  end
+
+  def print_nil(obj)
+    obj.present? ? obj : "none"
```

Comment:
suggesting code climate fix
```suggestion
    obj.presence || "none"
```